MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```
@@ -107,12 +123,132 @@ async Task<MSBuildCompletionSessionContext> GetSessionContext (IAsyncCompletionS
+			var httpClient = new HttpClient ();
+			string result = null;
+			var searchQuery = rr.Reference.ToString ().ToLower ();
+			if (string.IsNullOrEmpty (searchQuery)) {
+				return CompletionContext.Empty;
+			}
+
+			try {
+				var cancellationPackage = new CancellationTokenSource (TimeSpan.FromSeconds (10));
+				var packageResponse = await httpClient.GetAsync ($"https://api-v2v3search-0.nuget.org/autocomplete?q={searchQuery}&skip=0&take=100", cancellationPackage.Token);
+				result = await packageResponse.Content.ReadAsStringAsync ();
+
+			} catch (Exception) {
+				return CompletionContext.Empty;
+			}
+
+			var packageJSONObject = JObject.Parse (result);
+			var packageArray = (JArray)packageJSONObject["data"];
+
+			foreach (var package in packageArray) {
+				var info = package.ToString ();
+				items.Add (CreateCompletionItem (new ItemInfo (info, "description 1", "description 2"),
+							doc, rr));
+			}
+			return new CompletionContext (ImmutableArray<CompletionItem>.Empty.AddRange (items));
+		}
+
+		private async Task<CompletionContext> GetPackageVersionsAsync (MSBuildRootDocument doc,
+																MSBuildResolveResult rr)
+		{
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			var items = new List<CompletionItem> ();
+			var httpClient = new HttpClient (); ;
+			string result = null;
+			var packageName = rr.XElement.Attributes.First.Value.ToLower ();
+
+			if (string.IsNullOrEmpty (packageName)) {
+				return CompletionContext.Empty;
+			}
+
+			try {
+				//includes pre-release: https://docs.microsoft.com/en-us/nuget/api/search-query-service-resource
+				var cancellationVersion = new CancellationTokenSource (TimeSpan.FromSeconds (100));
+				var versionResponse = await httpClient.GetAsync ($"https://api-v2v3search-0.nuget.org/query?q={packageName}&prerelease=true", cancellationVersion.Token);
+				result = await versionResponse.Content.ReadAsStringAsync ();
+
+			} catch (Exception) {
```

In cases like this I would suggest logging the exception instead of silently swallowing it.